src/main/java/org/example/handlers/ExchangeRateAPIHandler.java

```diff
@@ -0,0 +1,67 @@
+package org.example.handlers;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.example.JSONMapper;
+import org.example.models.CurrencyExchange;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.LocalDate;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class ExchangeRateAPIHandler {
+    private static final Logger logger = Logger.getLogger(ExchangeRateAPIHandler.class.getName());
+
+    private final HttpClient client;
+    private static final URI API_URL = URI.create("https://api.nbp.pl/api/exchangerates/");
```

Comment:
Could be moved to a separete class

Reply:
It's good idea, URI was moved to APIConstans class